### PR TITLE
Update all statuses 

### LINKED ISSUE #2012: Review and change statuses for operations
Currently, during an upgrade operation, the statuses follow:
- not_deployed
- deploying
- action_succeeded

None of these statuses are ideal and do not really represent what is going on. Same for deleting. Statuses need to reflect what the operation is intended to do.

Following this, the UI will need a small update to add `successStates` and `failedStates` etc.

### LINKED ISSUE #2146: E2E tests continue even on failure
The e2e tests are only asserting for a single success / failure status, whereas there are a number of failure statuses. This causes the tests to continue even after failure. 

Note that #2012 should go in first as all the statuses are changing.

### PR DESCRIPTION
**Pull request by damoodamoo**
# Closes #2012 and Closes #2146

Fixes the status lifecycles to make more sense to the user based on the action they're performing.

### CODE REVIEW

File: api_app/models/domain/operation.py [hunk 1]
```diff
@@ -74,7 +84,7 @@ class Operation(AzureTREModel):
     resourceId: str = Field(title="resourceId", description="GUID identifying the resource")
     resourcePath: str = Field(title="resourcePath", description="Path of the resource undergoing change, i.e. '/workspaces/guid/workspace-services/guid/'")
     resourceVersion: int = Field(0, title="resourceVersion", description="Version of the resource this operation relates to")
-    status: Status = Field(Status.NotDeployed, title="Operation status")
+    status: Status = Field(None, title="Operation status")
```
[the remark this answers is not in the document]
Comment (damoodamoo) [reply api_app/models/domain/operation.py:87]: yea I think so, we have unit test coverage of this so I think we're ok

File: api_app/models/domain/operation.py [hunk 2]
```diff
@@ -13,19 +13,27 @@ class Status(str, Enum):
...
+
+    AwaitingAction = strings.RESOURCE_STATUS_AWAITING_ACTION
     InvokingAction = strings.RESOURCE_ACTION_STATUS_INVOKING
     ActionSucceeded = strings.RESOURCE_ACTION_STATUS_SUCCEEDED
     ActionFailed = strings.RESOURCE_ACTION_STATUS_FAILED
-    PipelineDeploying = strings.RESOURCE_ACTION_STATUS_PIPELINE_DEPLOYING
-    PipelineFailed = strings.RESOURCE_ACTION_STATUS_PIPELINE_FAILED
-    PipelineSucceeded = strings.RESOURCE_ACTION_STATUS_PIPELINE_SUCCEEDED
+
+    PipelineRunning = strings.RESOURCE_ACTION_STATUS_PIPELINE_RUNNING
```
[the remark this answers is not in the document]
Comment (damoodamoo) [reply api_app/models/domain/operation.py]: yup good idea!

**Comment by damoodamoo** [comment]
/test-force-approve - tests have run but not reported for some reason

**Review by tanya-borisova — APPROVED** [review]

**Pull request merged**